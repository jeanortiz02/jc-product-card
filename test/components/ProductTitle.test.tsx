
import React from 'react';
import { render } from '@testing-library/react';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/product';


product1


describe('ProductTitle', () => {
    
    test('debe mostrar el componente correctamente con el titulo personalizado', () => {
        
        const { container } = render(
            <ProductTitle title="Custom Product"/>
        )

       expect(container).toMatchSnapshot();
        
    });

    test('Debe de mostrar el componente con el nombre de producto', () => {
        const { container } = render(
            <ProductCard
                product={product1}

            >
                {
                    ()=> (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        )

       expect(container).toMatchSnapshot();
    });
});
  