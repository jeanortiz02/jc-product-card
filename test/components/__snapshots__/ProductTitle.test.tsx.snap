// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe de mostrar el componente con el nombre de producto 1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <span
      class="productDescription undefined"
    >
      Coffe Mug - Sin imagen
    </span>
  </div>
</div>
`;

exports[`ProductTitle debe mostrar el componente correctamente con el titulo personalizado 1`] = `
<div>
  <span
    class="productDescription undefined"
  >
    Custom Product
  </span>
</div>
`;
